halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Providers.java

```diff
@@ -54,4 +58,28 @@ public void accept(ProblemSetBuilder psBuilder, Validator v) {
   public NodeReference getReference() {
     return parent.getReference();
   }
+
+  public static Class<? extends Provider> translateProviderType(String providerName) {
+    Optional<? extends Class<?>> res = Arrays.stream(Providers.class.getDeclaredFields())
+        .filter(f -> f.getName().equals(providerName))
+        .map(Field::getType)
+        .findFirst();
+
+    if (res.isPresent()) {
+      return (Class<? extends Provider>)res.get();
+    } else {
+      throw new IllegalArgumentException("No provider with name \"" + providerName + "\" handled by halyard");
```

Comment:
can you list what you do know?
This is extensible so hopefully get it from somewhere else.